lib/common/socket/evloop/epoll.c.h

```diff
@@ -196,9 +237,34 @@ int evloop_do_proceed(h2o_evloop_t *_loop, int32_t max_wait)
     if (update_status(loop) != 0)
         return -1;
 
-    /* poll */
+        /* poll */
+#ifdef H2O_HAS_YNL_H
+    /* save a few syscalls if epoll bp params are unchanged */
+    if (loop->super.bp.epoll_bp_changed) {
+        evloop_set_bp(loop, loop->super.bp.epoll_bp_usecs, loop->super.bp.epoll_bp_budget, loop->super.bp.prefer_busy_poll);
+        loop->super.bp.epoll_bp_changed = false;
+    }
+
+    if (loop->super.bp.epoll_bp_usecs) {
+        uint64_t _time = loop->super.bp.epoll_bp_usecs;
+        time_t seconds = 0;
+        long int nsec = 0;
+
+        if (!loop->super.bp.epoll_nonblock) {
+            seconds = _time / 1000000;
+            nsec = (_time % 1000000) * 1000;
+        }
+
+        struct timespec ts = {.tv_sec = seconds, .tv_nsec = nsec};
+        nevents = epoll_pwait2(loop->ep, events, sizeof(events) / sizeof(events[0]), &ts, NULL);
+    } else {
+        nevents = epoll_wait(loop->ep, events, sizeof(events) / sizeof(events[0]), max_wait);
+    }
+#else
     max_wait = adjust_max_wait(&loop->super, max_wait);
```

Comment:
i think `max_wait` needs to be moved up because it is used in the else case on line 261 above, but not computed.

Reply:
in the case of `suspend`, i think there should be extra logic added to set `max_wait = -1` -- in `suspend` mode, the idea is that the polling happens in the kernel until an RX queue finds nothing, at which point IRQs are re-enabled and the process can be put to sleep (because there is no data to process). and it will be re-awoken when data arrives and an IRQ is generated.

so, in psuedocode maybe something like:

```
if (mode == suspend)
  max_wait = -1
else
  max_wait = adjust_max_wait(&loop->super, max_wait);
```

or maybe move that logic into `adjust_max_wait` itself and have it return `-1` if suspend mode is active for this `loop` ?

Reply:
Thank you addressed in 12b8858938581e576acabbf7b89208a9deb4e08d